record-and-playback/core/scripts/post_events/post_events_analytics.rb.example

```diff
@@ -0,0 +1,203 @@
+# any later version.
+#
+# BigBlueButton is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with BigBlueButton; if not, see <http://www.gnu.org/licenses/>.
+#
+
+require 'net/http'
+require "jwt"
+require 'bbbevents'
+require 'socket'
+require "trollop"
+require File.expand_path('../../../lib/recordandplayback', __FILE__)
+
+def send_data(analytics_url, secret, payload)
+  # Setup a token that expires in 24hrs
+  exp = Time.now.to_i + 24 * 3600
+  exp_payload = { :exp => exp }
+  token = JWT.encode exp_payload, secret, 'HS512', { typ: 'JWT' }
+
+  uri = URI.parse(analytics_url)
+  http = Net::HTTP.new(uri.host, uri.port)
+  http.use_ssl = (uri.scheme == 'https')
+
+  # Setup a request and attach our JWT token
+  request = Net::HTTP::Post.new(uri.request_uri, {
+    'Content-Type' => 'application/json',
+    'Authorization' => "Bearer #{token}",
+    'User-Agent' => 'BigBlueButton Analytics Callback'
+  })
+
+  # Send out data as json body
+  request.body = payload.to_json
+
+  # Display debug results
+  #http.set_debug_output($stdout) 
+
+  response = http.request(request)
+  code = response.code.to_i
+  puts response.body
+
+  if code < 200 || code >= 300
+    puts "Failed when calling #{uri.request_uri}"
+  else
+    puts "Success"
+  end
+end
+
+def format_analytics_data!(data)
+  tmp_metadata = data["metadata"]
+  array_keys = tmp_metadata.keys
+  array_keys.each { |item|
+    if item.start_with?("gl_")
+      tmp_metadata = tmp_metadata.tap { |hs|
+        hs.delete(item) 
+      }
+    end
+  }
+
+  # Remove meeting id as we don't want it passed to 3rd party inside metadata
+  tmp_metadata.delete("meeting_id")
+
+  # Remove the internal meeting id generated by bbbevents gem
+  data = data.tap { |hs|
+    hs.delete("meeting_id") 
+  }
+
+  # Convert CamelCase keys to snake_keys. This is done in bbbevents gem
+  # but we do it here too anyways.
+  tmp_metadata.deep_transform_keys! do |key|
+    k = key.to_s.underscore rescue key
+    k.to_sym rescue key
+  end       
+
+  # Remove all internal user ids
+  attendees = data["attendees"]
+  attendees.each { |attendee|
+    attendee.delete("id")
+  }
+
+end
+
+opts = Trollop::options do
+  opt :meeting_id, "Meeting id to archive", :type => String
+end
+meeting_id = opts[:meeting_id]
+
+i_got_this = false
+if not i_got_this
+  BigBlueButton.logger.info("######################################################")
+  BigBlueButton.logger.info("!!! Please read setup instructions in bottome of this script. !!!")
+  BigBlueButton.logger.info("######################################################")
+  exit 0
+end
+
+
+logger = Logger.new(STDOUT)
+#logger = Logger.new("/var/log/bigbluebutton/post_events.log", 'weekly' )
```

Comment:
It would be good to have the logs in a log file.

Reply:
Wouldn't be the same we use for the other post scripts?

`logger = Logger.new("/var/log/bigbluebutton/post_publish.log", 'weekly' )`